client/components/data/query-user-devices/index.js

```diff
@@ -0,0 +1,37 @@
+/**
+ * External dependencies
+ */
+import { PropTypes, Component } from 'react';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import { requestUserDevices } from 'state/user-devices/actions';
+import { isRequestingUserDevices } from 'state/selectors';
+
+class QueryUserDevices extends Component {
+	static propTypes = {
+		isRequesting: PropTypes.bool,
+		requestUserDevices: PropTypes.func
+	}
+
+	componentWillMount() {
```

Comment:
Let's use `componentDidMount` for network requests. This is mainly to guard against future React updates.

Also see: https://gist.github.com/bvaughn/923dffb2cd9504ee440791fade8db5f9